cluster/cluster.go

```diff
@@ -79,20 +79,31 @@ func Start() {
 	log.Info("CLU Start: joined to %d nodes in cluster", n)
 }
 
+type partitionCandidates struct {
+	priority int
+	nodes    []Node
+}
+
 // return the list of nodes to broadcast requests to
-// Only 1 member per partition is returned. This list includes
-// ThisNode if it is capable of handling queries.
+// Only 1 member per partition is returned.
+// The nodes are selected based on priority, using a random selection
+// if there are multi nodes with the highest priority.
```

Comment:
please don't remove the comment explaining that ThisNode can be included. that's really good to know. Also I don't think "Only 1 member per partition is returned." is only correct for symmetric partition assignments, not for shuffled assignments. that's how we have it setup so for us it's true, but it doesn't look like an intrinsic property of the code to be that way. so let's clarify